TechRadarApi.Tests/V1/UseCase/DeleteTechnologyByIdUseCaseTests.cs

```diff
@@ -0,0 +1,52 @@
+using System.Linq;
+using AutoFixture;
+using TechRadarApi.V1.Boundary.Response;
+using TechRadarApi.V1.Domain;
+using TechRadarApi.V1.Factories;
+using TechRadarApi.V1.Gateways;
+using TechRadarApi.V1.UseCase;
+using FluentAssertions;
+using Moq;
+using Xunit;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System;
+using Microsoft.AspNetCore.Mvc;
+
+namespace TechRadarApi.Tests.V1.UseCase
+{
+    public class DeleteTechnologyByIdUseCaseTests
+    {
+        private Mock<ITechnologyGateway> _mockGateway;
+        private DeleteTechnologyByIdUseCase _classUnderTest;
+        private Fixture _fixture;
+
+        public DeleteTechnologyByIdUseCaseTests()
+        {
+            _mockGateway = new Mock<ITechnologyGateway>();
+            _classUnderTest = new DeleteTechnologyByIdUseCase(_mockGateway.Object);
+            _fixture = new Fixture();
+        }
+
+        [Fact]
+        public async Task DeleteTechnologyByIdUseCaseReturnsNullWhenTechnologyIsNotFound()
+        {
+            var query = _fixture.Build<TechnologyResponseObject>().With(x => x.Id).Create();
+
+            var response = await _classUnderTest.Execute(query.Id).ConfigureAwait(false);
+            response.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task DeleteTechnologyByIdUseCaseReturnsOkResponse()
+        {
+            var technology = _fixture.Create<Technology>();
+            var query = _fixture.Build<TechnologyResponseObject>().With(x => x.Id).Create();
+
+            _mockGateway.Setup(x => x.DeleteTechnologyById(technology));
+
+            var response = await _classUnderTest.Execute(query.Id).ConfigureAwait(false);
+            response.Should().BeOfType(typeof(NoContentResult));
```

Comment:
Instead of checking the response type we should check the object of the response is as expected